client/src/details/details.cpp

```diff
@@ -473,3 +485,42 @@ void Details::setItemsTreeModel(ItemsTreeModel *model)
     // keep the member - could be useful later
     mItemsTreeModel = model;
 }
+
+void Details::copyAddressFromGroup(QGroupBox *box)
+{
+    QMap<QString, QString> fieldNames;
+    fieldNames.insert(QStringLiteral("city"), {});
+    fieldNames.insert(QStringLiteral("country"), {});
+    fieldNames.insert(QStringLiteral("postalcode"), {});
+    fieldNames.insert(QStringLiteral("state"), {});
+    fieldNames.insert(QStringLiteral("street"), {});
+    fillAddressFielsMap(box, fieldNames);
+
+    QMap<QString, QString> address;
+    QMutableMapIterator<QString, QString> i(fieldNames);
+    while (i.hasNext()) {
+        i.next();
+        QWidget *field = box->findChild<QWidget*>(i.value());
+        if (field) {
+            QPlainTextEdit *plainTextEdit = qobject_cast<QPlainTextEdit*>(field);
+            if (plainTextEdit) {
+                address.insert(i.key(), plainTextEdit->toPlainText());
+                continue;
+            }
+            QLineEdit *lineEdit = qobject_cast<QLineEdit*>(field);
+            if (lineEdit) {
+                address.insert(i.key(), lineEdit->text());
+                continue;
+            }
+
+            address.insert(i.key(), {});
```

Comment:
This line doesn't serve any purpose. You use value() below, so you'll get an empty string anyway, for missing fields.
How can it be missing, though? Shouldn't this be a qWarning? It would mean the page has given us a field name for which we found an unsupported widget type (or no widget, if you add an `else` branch)...